### PR TITLE
Task 5: Get Articles

### PR DESCRIPTION
**Pull request by ww1659**

### CODE REVIEW

**Comment by ww1659** [comment]
Sorry - not finished! Haven't added the comments.

File: __tests__/app.test.js [hunk 1]
```diff
@@ -48,13 +41,50 @@ describe("api/topics", () => {
...
+        articles.forEach((article) => {
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+          expect(typeof article.comment_count).toBe("string");
+          expect(article.hasOwnProperty("body")).toBe(false);
```
Comment (Poonam-raj) [inline __tests__/app.test.js:62]: lovely assertions here!

**Comment by ww1659** [comment]
Added the comment_count now!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -48,13 +41,50 @@ describe("api/topics", () => {
...
+        });
+      });
+  });
+  test("GET:404 returns status 404 for a non-existent path", () => {
+    return request(app).get("/api/article").expect(404);
```
Comment (Poonam-raj) [inline __tests__/app.test.js:67]: would prefer you to also expect a message for this error too. Later on we will be making more 404 errors and we want to know what kind of 404 these are so the message will help us differentiate it

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Edit: I've just seen you've added in comment_count I'll review the changes

File: models/articles-model.js [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+exports.fetchArticles = () => {
+  const fetchArticleQuery = `
+  SELECT a.article_id, title, topic, a.author, a.created_at, a.votes, article_img_url, COUNT(comment_id) as comment_count
+  FROM articles as a
+  LEFT JOIN comments as c
+  ON a.article_id = c.article_id
+  GROUP BY a.article_id, title, topic, a.author, a.created_at, a.votes, article_img_url
+  ORDER BY a.created_at DESC;`;
+
+  return db.query(fetchArticleQuery).then((result) => {
```
Comment (Poonam-raj) [inline models/articles-model.js:12]: lovelyyy

**Review by Poonam-raj — APPROVED** [review]
smashing work, just left a note about that 404 - feel free to make that change if it makes sense to you. but you're good to merge

**Pull request merged**